crates/uv-cli/src/lib.rs

```diff
@@ -2087,6 +2087,10 @@ pub struct ToolRunArgs {
     /// - `/home/ferris/.local/bin/python3.10` uses the exact Python at the given path.
     #[arg(long, short, env = "UV_PYTHON", verbatim_doc_comment)]
     pub python: Option<String>,
+
+    /// Internal flag indicating execution via the `uvx` binary.
+    #[arg(long, hide = true)]
+    pub invoked_via_uvx: bool,
```

Comment:
Instead of adding a flag, could you make use of the `Uvx` variant at https://github.com/astral-sh/uv/blob/c14de2a92a4a4797be56eca79bbe644c762aab7b/crates/uv/src/lib.rs#L599

We should just be able to tell if `uvx` was invoked from there with `matches!`

Reply:
Missed that before, done 😅

Reply:
I took this a bit further in https://github.com/astral-sh/uv/pull/5014/commits/3426d0ad4e4a68dc68753b38a798d4e4b7f9a856

Reply:
Thanks! very cool, didn't know about that syntax/feature before!